src/crdt.rs

```diff
@@ -551,6 +551,33 @@ impl Crdt {
         blob_sender.send(q)?;
         Ok(())
     }
+    /// FIXME: This is obviously the wrong way to do this. Need to implement leader selection
+    fn top_leader(&self) -> Option<PublicKey> {
+        let mut table = HashMap::new();
+        let def = PublicKey::default();
+        let cur = self.table.values().filter(|x| x.current_leader_id != def);
+        for v in cur {
+            let cnt = table.entry(&v.current_leader_id).or_insert(0);
+            //let cnt = table.get_mut(&v.current_leader_id).unwrap();
+            *cnt += 1;
+            println!("leader {:?} {}", &v.current_leader_id[..4], *cnt);
+        }
+        let mut sorted: Vec<_> = table.iter().collect();
+        sorted.sort_by_key(|a| a.1);
+        sorted.last().map(|a| *(*(*a).0))
+    }
+
+    /// FIXME: This is obviously the wrong way to do this. Need to implement leader selection
+    /// A t-shirt for the first person to actually use this bad behavior to attack the alpha testnet
+    fn update_leader(&mut self) {
+        if let Some(lid) = self.top_leader() {
```

Comment:
`id` would be a better name since `lid` is a word. If not specific enough, use `leader_id`